Manual/Iterators.lean

```diff
@@ -342,6 +369,44 @@ This iterator is useful with combinators such as {name}`Iter.zip`:
 2: pachycephalosaurus
 ```
 :::
+
+:::paragraph
+In contrast to the previous example, this loop terminates because `xs.iter` is a finite iterator,
```

Comment:
This sentence should be part of the preceding paragraph, right?

Also:
```suggestion
In contrast to {name}`f`, this loop terminates because `xs.iter` is a finite iterator.
```